pkg/api/service/scans.go

```diff
@@ -339,64 +354,49 @@ func (s ScansService) ProcessScanCheckNotification(ctx context.Context, msg []by
 		return nil
 	}
 
-	// Don't take into account inconsistent progress in a message with a terminal status.
-	if checkStates.IsTerminal(checkMssg.Status) && (checkProgress != 1.0) {
-		_ = level.Error(s.logger).Log("FixingInvalidProgressInTerminalStatus", checkProgress, "Status", checkMssg.Status)
-		checkProgress = 1
-		checkMssg.Progress = &checkProgress
+	_, err = s.db.UpsertCheck(scanID, checkID, checkMssg, api.CheckStates.LessOrEqual(checkMssg.Status))
+	if err != nil {
+		return err
+	}
+	// As a check message does not contain all the information
+	// of a check we must merge with the the info of the check in the DB.
+	check := mergeChecks(dbCheck, checkMssg)
+	if err != nil {
+		return err
+	}
+	programID := ""
+	if check.Tag != nil {
+		programID = *check.Tag
 	}
```

Comment:
Why are we updating the programID with the tag? Shouldn't we use the program ID that is set at the DB level for the check?

Reply:
Yep, that's what we are doing!! I mean the program ID is indeed the Tag field stored in the db for a check, so we can use it here: ```s.notifyCheck(check, programID)```  without having to get the scan from the DB. That means that for the check messages that are not terminal we don't need to go to the db to get the scan data just for getting the program id(that's what we were doing previously)

Reply:
Sorry you were right!!, the tag is not the program exactly, so it does not work properly!! I'm changing it